### PR TITLE
create resource padding option

### LINKED ISSUE #349: create resource padding option
```typescript
scaleBand()
      .range([0, resourceWidth])
      .paddingInner(spacing)
      .domain(this.resources);
```
If there any issues with the grid lines, create another ticket to address them

### PR DESCRIPTION
**Pull request by wflemingnz-retired**
#closes #349

### CODE REVIEW

File: projects/ngx-d3timeline/src/lib/scales/scale-utils.ts [hunk 1]
```diff
@@ -37,11 +37,15 @@ export function rescaleTime(
...
-  orientation: Orientation
+  orientation: Orientation,
+  lateralPadding: number
 ): BandScale {
   return scaleBand()
     .domain(getBandScaleDomain(activities))
-    .range(getRange(view, orientation));
+    .range(getRange(view, orientation))
+    .paddingInner(lateralPadding)
+    .paddingOuter(lateralPadding)
```
Comment (umarmohammed) [inline projects/ngx-d3timeline/src/lib/scales/scale-utils.ts]: not sure we need outer padding

File: projects/ngx-d3timeline/src/lib/options/options.ts [hunk 2]
```diff
@@ -22,5 +27,8 @@ export const defaultOptions: CompleteOptions = {
   resourceAxis: {
     tickLineLength: 0,
     showGridLines: true
+  },
+  activities: {
+    lateralPadding: 0.25
```
Comment (umarmohammed) [inline projects/ngx-d3timeline/src/lib/options/options.ts]: isn't this just resourcePadding? Not sure it should be part of activityOptions

**Review by umarmohammed — CHANGES_REQUESTED** [review]

File: projects/ngx-d3timeline/src/lib/options/options.ts (hunk 2 above)
Comment (wflemingnz-retired) [reply projects/ngx-d3timeline/src/lib/options/options.ts]: My thinking was that you could specify it at an activity(type) level as well as a global level. So Driving could be padded more than Handling etc. But now I don't think that is achievable via scale.paddingInner/Outer as there is only one scale. You had something else in your POC I think maxWidth but I didn't get my head around how the two interact, it seems like maybe they were serving a similar purpose. Probably best to have a chat about this one.

File: projects/ngx-d3timeline/src/lib/scales/scale-utils.ts (hunk 1 above)
Comment (wflemingnz-retired) [reply projects/ngx-d3timeline/src/lib/scales/scale-utils.ts]: Its required to get even spacing on the edge resources. But see comment below maybe I'm misusing it.

With outer padding:
image

Without outer padding:
image

Comment (wflemingnz-retired) [reply projects/ngx-d3timeline/src/lib/scales/scale-utils.ts]: removed outer padding

File: projects/ngx-d3timeline/src/lib/options/options.ts (hunk 2 above)
Comment (wflemingnz-retired) [reply projects/ngx-d3timeline/src/lib/options/options.ts]: moved to options.resource.gap

**Review by umarmohammed — APPROVED** [review]

**Pull request merged**